core/commands/partials/terraform_digitalocean_generator.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+require 'date'
+require 'erb'
+require 'socket'
+require_relative '../../models/result'
+
+# The class generates the Terraform infrastructure file for Digital Ocean provider
+class TerraformDigitaloceanGenerator
+  # Initializer.
+  # @param configuration_id [String] configuration id
+  # @param digitalocean_config [Hash] hash of Digital Ocean configuration
+  # @param logger [Out] logger
+  # @param configuration_path [String] path to directory of generated configuration
+  # @param ssh_keys [Hash] ssh keys info in format { public_key_value, private_key_file_path }
+  # @return [Result::Base] generation result.
+  def initialize(configuration_id, digitalocean_config, logger, configuration_path, ssh_keys)
+    @configuration_id = configuration_id
+    @digitalocean_config = digitalocean_config
+    @ui = logger
+    @configuration_path = configuration_path
+    @configuration_tags = { configuration_id: @configuration_id }
+    @public_key_value = ssh_keys[:public_key_value]
+    @private_key_file_path = ssh_keys[:private_key_file_path]
+  end
+
+  # Generate a Terraform configuration file.
+  # @param node_params [Array<Hash>] list of node params.
+  # @param configuration_file_path [String] path to generated Terraform infrastructure file.
+  # @return [Result::Base] generation result.
+  def generate_configuration_file(node_params, configuration_file_path)
+    raise 'Digital Ocean is not configured' if @digitalocean_config.nil?
+
+    file = File.open(configuration_file_path, 'w')
+    file.puts(file_header)
+    file.puts(provider_resource)
+    node_params.each do |node|
+      print_node_info(node)
+      file.puts(generate_node_definition(node))
+    end
+    file.close
+  rescue StandardError => e
+    Result.error(e.message)
+  else
+    Result.ok('')
+  end
+
+  # Generate the instance name.
+  # @param configuration_id [String] configuration id.
+  # @param node_name [String] name of the node.
+  # @return [String] generated instance name.
+  def self.generate_instance_name(configuration_id, node_name)
+    "#{configuration_id}-#{format_string(node_name)}"
+  end
+
+  # Format string (only letters, numbers and hyphen).
+  # @param string [String] string for format
+  # @return [String] formatted string.
+  def self.format_string(string)
```

Comment:
Should we put this into the TerraformService as such transition is common to the Terraform?